### PR TITLE
fuse: Parse OSX fuse version properly

### LINKED ISSUE #3410: Is Fuse 3.x.x supported? If yes, I'm getting error
The fuse version check tries to match with `2.7.x` or `2.8.x`. Latest stable version is `3.5.3` and I get this error:

```
Error: OSXFUSE version 3.5.3 not supported.

OSXFUSE versions <2.7.2 are known to cause kernel panics!
Please upgrade to the latest OSXFUSE version.
It is recommended you install it from the OSXFUSE website:

	http://osxfuse.github.io/

For more help, see:

	https://github.com/ipfs/go-ipfs/issues/177
```
From [mount_darwin.go:120](https://github.com/ipfs/go-ipfs/blob/73cd8b3e98aba252f0eadcc625472103a2dd1d53/fuse/node/mount_darwin.go#L141):

```
if strings.HasPrefix(ov, "2.7.") || strings.HasPrefix(ov, "2.8.") {
	return nil
}
```
#### Version information:
go-ipfs version: 0.4.4-
Repo version: 4
System version: amd64/darwin
Golang version: go1.7

#### Type: Bug

### PR DESCRIPTION
**Pull request by Kubuxu**
Resolves #3410
License: MIT
Signed-off-by: Jakub Sztandera <kubuxu@protonmail.ch>

### CODE REVIEW

**Comment by VictorBjelkholm** [comment]
Tried it out with osx-fuse 3.5.3, now passing the version check but I can't get FUSE to work properly anymore. Not sure IPFS is compatible with latest osx-fuse, gonna have to let someone else to check as well.

Logs
```
15:05:18.280 ERROR  fuse/ipns: looking up /ipns/QmaM5uS4D4ujbL6euLKRc9d74wULSFKmWaPsBU2yzuzD7i: Could not resolve name. ipns_unix.go:99
15:05:18.280 ERROR       node: error mounting: fork/exec /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs: no such file or directory mount_unix.go:97
15:05:18.280 ERROR       node: error mounting: Could not resolve name. mount_unix.go:101
Error: fork/exec /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs: no such file or directory
```

**Comment by Kubuxu** [comment]
Hmm, the first and last messages look like long existing bug, you can resolve it by running:
`ipfs name publish QmbFMke1KXqnYyBBWxB74N4c5SBnJMVAiMNRcGu6x1AwQH`

**Comment by VictorBjelkholm** [comment]
@Kubuxu yeah, that fixed the resolve error but I'm still getting this:

```
15:13:57.868 ERROR       node: error mounting: fork/exec /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs: no such file or directory mount_unix.go:97
15:13:57.869 ERROR       node: error mounting: fork/exec /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs: no such file or directory mount_unix.go:101
Error: fork/exec /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs: no such file or directory
```

**Comment by Kubuxu** [comment]
@whyrusleeping this should be ready for CR/merge.

File: fuse/node/mount_darwin.go [hunk 1]
```diff
@@ -54,6 +56,8 @@ It is recommended you install it from the OSXFUSE website:
 For more help, see:
 
 	https://github.com/ipfs/go-ipfs/issues/177
+
+OSXFUSE versions >3.0.0 are not compatible with version of FUSE library in current use.
```
Comment (whyrusleeping) [inline fuse/node/mount_darwin.go]: "This will be fixed in the near future, see issue number XYZ..."

File: package.json [hunk 2]
```diff
@@ -293,6 +293,12 @@
       "hash": "QmXGevGDVTqeKdisBzaxEK4CJZqfxeXiVSWLaXaVWcG5on",
       "name": "go-smux-multiplex",
       "version": "1.1.4"
+	},
+	{
+      "author": "blang",
+      "hash": "QmU1N5xVAUXgo3XRTt6GhJ2SuJEbxj2zRgMS7FpjSR2U83",
+      "name": "semver",
```
Comment (whyrusleeping) [inline package.json]: do we not already have a semver package being used?

Comment (Kubuxu) [inline package.json]: No, but you use this one in gx source code.

**Comment by Kubuxu** [comment]
@whyrusleeping updated

**Pull request merged**